maintenance_equipment_scrap/models/maintenance_equipment.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = 'maintenance.equipment'
+
+    equipment_scrap_template_id = fields.Many2one(
+        'mail.template',
+        string='Equipment Scrap Email Template',
+        default=(lambda self:
+                 self.env.user.company_id.equipment_scrap_template_id)
+    )
+
+    @api.multi
+    def action_perform_scrap(self):
+        self.ensure_one()
+        action = self.env.ref(
+            'maintenance_equipment_scrap.wizard_perform_equipment_scrap_action'
+        )
+        result = action.read()[0]
+        return result
```

Comment:
This module "actively" only does two things:
- Sends an email when a scrap is performed, and only if on the equipment is selected a scrapping template;
- Writes the scrap_date field when a scrap is performed.

Still, I think that what really makes this module useful is that it provides a structure for the equipment scrapping functionality. To better explain, consider that in base Odoo, the equipment scrapping functionality just doesn't exist: as far as I can see, an equipment is considered scrapped when the user manually set a scrap date; there is no state change, no active/inactive switch, nothing else that says that a given equipment is scrapped. 
Because of this lack of structure, it's quite tricky to define additional logics to be ran when an equipment is scrapped (such logic as, for exemple, to send an email when it occurs).

This module makes the scrap_date field read-only and provides a procedure for a user to scrap an equipment: it has to click the "scrap" button, to fill a form and finally to submit it. This provides, apart from the obvious functionality of sending an email, an entry-point for adding more and more logic (making the equipment inactive is a possible example, as it could be making some changes on the related asset, or whatever else we can think about, and that could be added in this or in other, more specific-purpose, modules).

What do you think about that?